core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -944,6 +944,10 @@ public XContentBuilder dateValueField(XContentBuilderString rawFieldName, XConte
         return this;
     }
 
+    public XContentBuilder timeValueField(String rawFieldName, String readableFieldName, TimeValue timeValue) throws IOException {
```

Comment:
Maybe is it as simple as changing `rawFieldName` to `millisFieldName` and documenting the method properly?